packages/js/src/executors/release-publish/release-publish.impl.ts

```diff
@@ -56,32 +58,25 @@ export default async function runExecutor(
     };
   }
 
-  const npmPublishCommandSegments = [`npm publish --json`];
+  const { registry, tag, registryConfigKey } = await parseRegistryOptions(
+    context.root,
+    {
+      root: packageRoot,
+      manifest: projectPackageJson,
+      manifestPath: packageJsonPath,
+    },
+    {
+      registry: options.registry,
+      tag: options.tag,
+    }
+  );
+
   const npmViewCommandSegments = [
-    `npm view ${packageName} versions dist-tags --json`,
+    `npm view ${packageName} versions dist-tags --json --"${registryConfigKey}=${registry}"`,
```

Comment:
When there's a scoped package, the only way to override a scoped package config in `.npmrc` is to use the scope in the registry argument. For example:

`npm view @scope/pkg1 versions dist-tags --json --"@scope:registry=http://localhost:4873"`